web/src/components/navbar/NavBarPopupMenu.tsx

```diff
@@ -0,0 +1,192 @@
+import { NavMenuItem } from "@/components/navbar/NavMenuItem";
+import { Icon } from "@/components/njwds/Icon";
+import { AuthAlertContext } from "@/contexts/authAlertContext";
+import { AuthContext } from "@/contexts/authContext";
+import { triggerSignIn } from "@/lib/auth/sessionHelper";
+import { onSelfRegister, onSignOut } from "@/lib/auth/signinHelper";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { getNavBarBusinessTitle } from "@/lib/domain-logic/getNavBarBusinessTitle";
+import { ROUTES } from "@/lib/domain-logic/routes";
+import analytics from "@/lib/utils/analytics";
+import { getUserNameOrEmail } from "@/lib/utils/helpers";
+import Config from "@businessnjgovnavigator/content/fieldConfig/config.json";
+import { MenuItem, MenuList } from "@mui/material";
+import { useRouter } from "next/router";
+import { ReactElement, useContext } from "react";
+
+export type MenuConfiguration =
+  | "profile"
+  | "profile-register-login"
+  | "profile-myNJ-logout"
+  | "login"
+  | "login-getstarted";
+
+export interface Props {
+  handleClose: () => void;
+  open?: boolean;
+  hasCloseButton?: boolean;
+  menuConfiguration: MenuConfiguration;
+}
+
+export const NavBarPopupMenu = (props: Props): ReactElement => {
+  const { userData, updateQueue } = useUserData();
+  const { dispatch } = useContext(AuthContext);
+  const { setRegistrationAlertStatus } = useContext(AuthAlertContext);
+
+  const router = useRouter();
+
+  const guestOrUserName =
+    props.menuConfiguration === "login"
+      ? Config.navigationDefaults.navBarGuestText
+      : getUserNameOrEmail(userData);
+  const isDashboardSelected = router.route === ROUTES.dashboard;
+  const isProfileSelected = router.route === ROUTES.profile;
+
+  const navBarBusinessTitle = getNavBarBusinessTitle(userData);
+
+  function handleListKeyDown(event: React.KeyboardEvent): void {
+    if (event.key === "Tab") {
+      event.preventDefault();
+      props.handleClose();
+    }
+  }
+
+  const loginMenuItem = (): ReactElement => {
+    return NavMenuItem({
+      onClick: (): void => {
+        analytics.event.landing_page_navbar_log_in.click.go_to_myNJ_login();
+        triggerSignIn();
+      },
+      icon: <ButtonIcon svgFilename="login" sizePx="25px" />,
+      itemText: Config.navigationDefaults.logInButton,
+      key: "loginMenuItem",
+    });
+  };
+
+  const logoutMenuItem = (): ReactElement => {
+    return NavMenuItem({
+      onClick: (): void => {
+        onSignOut(router.push, dispatch);
+        props.handleClose();
+      },
+      icon: <ButtonIcon svgFilename="logout" sizePx="25px" />,
+      itemText: Config.navigationDefaults.logoutButton,
+      key: "logoutMenuItem",
+    });
+  };
+
+  const myNjMenuItem = (): ReactElement => {
+    return NavMenuItem({
+      onClick: (): void => {
+        analytics.event.account_menu_myNJ_account.click.go_to_myNJ_home();
+        window.open(process.env.MYNJ_PROFILE_LINK || "", "_ blank");
```

Comment:
So oddly enough this is the code thats in production for this line and it works as expected by opening a new tab but the "_ blank" suprises me as I thought it would be "_blank"